drivers/storage/class/ramdisk/ramdisk.c

```diff
@@ -146,6 +146,21 @@ QueryParameters(IN PUNICODE_STRING RegistryPath)
     MinimumViewLength = 0x10000u;
     DefaultViewLength = 0x100000u;
 
+    OBJECT_ATTRIBUTES objAttrs;
```

Comment:
Why is it necessary to force opening the registry by handle here, instead of keeping the opening by name as was done in the original RtlQueryRegistryValues call below ? (that looked cleaner)

Reply:
I needed some routine, which takes `UNICODE_STRING` as an input and not `PWCHAR`.
I haven't found any info about guarantees here, that `RegistryPath` supplied to the `DriverEntry` function has to be a zero-ended string

Reply:
OK then the better would be to actually move the call to QueryParameters in DriverEntry **after** the step that creates such a copy of the registry string. You could even modify that copy of the registry string code to be sure it NULL-terminates the string always. This would allow to keep that QueryParameters function simple (as it was before) while ensuring that the registrypath UNICODE_STRING buffer is actually NULL-terminated.

Reply:
(Having or using a function similar to https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/input/i8042prt/i8042prt.c;hb=5099f24373d6f68f4b98241a3ea488595a109dc1#l504 )